charts/orchestrator/templates/_helpers.tpl

```diff
@@ -120,4 +120,25 @@
         {{- end -}}
         {{- .argoCDNamespace -}}
     {{- end -}}
+{{- end -}}
+
+{{- define "get-cluster-version" -}}
+  # Retrieve OCP version from `clusterversion` object
+  {{- $version :=(lookup "config.openshift.io/v1" "ClusterVersion" "" "version") }}
+  {{- range $version.status.history }}
+    {{- if eq .state "Completed" }}
+      {{- $v:=(semver .version) }}
+      {{- printf "OCP version is %d.%d" $v.Major $v.Minor }}
+    {{- end }}
+  {{- end }}
+
+  {{- $validMinors := list "13" "14" "15" "16" -}}
+  {{- if ne $v.Major "4" -}}
+      {{- fail (printf "Unsupported Major version: %s. Required: 4." $v.Major) -}}
+  {{- else if not (has $v.Minor $validMinors) -}}
+      {{- fail (printf "Unsupported Minor version: %s. Supported versions: 13, 14, 15, 16." $v.Minor) -}}
```

Comment:
What do you think of using the list to print its values instead of hardcoded ones?

Reply:
Especially since we already have it defined in line 136. I'll change.